smoke_test/steps/base_step.rb

```diff
@@ -0,0 +1,41 @@
+module SmokeTest
+  module Steps
+    class BaseStep
+      include Capybara::DSL
+
+      attr_accessor :state
+
+      def initialize(state)
+        puts "Step:#{step_name}"
+        @state = state
+      end
+
+      def validate!
+        true
+      end
+
+      def complete_step
+        fail NotImplementedError
```

Comment:
Yeah, that's spot on. Inheriting here is definitely a smell. Each step is being forced to have behaviour for the sake of a consistent interface when not all of the steps are the same. Most of the email related steps are really just validations - you can't progress, just make an assertion and do something else(another step).

:+1: The `with_retries` should definitely be extracted to a module and the printing should be hoisted up to the level that is controlling the steps.